container/checkpoint.go

```diff
@@ -1012,7 +983,189 @@ func CheckRuntime(current, expected string) bool {
 	return true
 }
 
-func runcCheckpointContainerd(ctx gocontext.Context, client *containerd.Client, task containerd.Task, opts ...CheckpointTaskOpts) (containerd.Image, error) {
+type ContainerdContainer struct {
+	containerd.Container
+	client *containerd.Client
+}
+
+func ContainerdCheckpoint(ctx context.Context, containerdClient *containerd.Client, id, ref string) error {
```

Comment:
should we add a flag or input here to differentiate b/ween rootfs containerd c/r and just containerd c/r?

Reply:
Honestly didn't add that because for full c/r I do not expect to use this outside of being used by orchestration. So a controller would c/r process state + rootfs. So I can rename this to a ContainerdRootfsCheckpoint or something similar